<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SpotifySettingsPage</class>
 <widget class="QWidget" name="SpotifySettingsPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>505</width>
    <height>853</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Spotify</string>
  </property>
  <layout class="QVBoxLayout" name="layout_spotifysettingspage">
   <item>
    <widget class="QCheckBox" name="enable">
     <property name="text">
      <string>Enable</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_warning">
     <property name="text">
      <string/>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
     <property name="margin">
      <number>10</number>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="credential_group">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Authentication</string>
     </property>
     <layout class="QFormLayout" name="layout_credential_group">
      <item row="0" column="0">
       <widget class="QLabel" name="label_username">
        <property name="text">
         <string>Username</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLineEdit" name="username"/>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_password">
        <property name="text">
         <string>Password</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="password">
        <property name="echoMode">
         <enum>QLineEdit::Password</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="button_login">
     <property name="text">
      <string>Login</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="LoginStateWidget" name="login_state" native="true"/>
   </item>
   <item>
    <widget class="QGroupBox" name="groupbox_preferences">
     <property name="title">
      <string>Preferences</string>
     </property>
     <layout class="QFormLayout" name="layout_preferences">
      <item row="0" column="0">
       <widget class="QLabel" name="label_quality">
        <property name="text">
         <string>Audio quality</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="quality"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_searchdelay">
        <property name="text">
         <string>Search delay</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="searchdelay">
        <property name="suffix">
         <string>ms</string>
        </property>
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
        <property name="singleStep">
         <number>50</number>
        </property>
        <property name="value">
         <number>1500</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_artistssearchlimit">
        <property name="text">
         <string>Artists search limit</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="artistssearchlimit">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_albumssearchlimit">
        <property name="text">
         <string>Albums search limit</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="albumssearchlimit">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_songssearchlimit">
        <property name="text">
         <string>Songs search limit</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QSpinBox" name="songssearchlimit">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QCheckBox" name="checkbox_download_album_covers">
        <property name="text">
         <string>Download album covers</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QCheckBox" name="checkbox_fetchalbums">
        <property name="text">
         <string>Fetch entire albums when searching songs</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="label_coversize">
        <property name="text">
         <string>Album cover size</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QComboBox" name="coversize"/>
      </item>
      <item row="8" column="0">
       <widget class="QCheckBox" name="checkbox_album_explicit">
        <property name="text">
         <string>Append explicit to album title for explicit albums</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="spacer_middle">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>30</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="layout_bottom">
     <item>
      <spacer name="spacer_bottom">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="label_spotify">
       <property name="minimumSize">
        <size>
         <width>64</width>
         <height>64</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>64</width>
         <height>64</height>
        </size>
       </property>
       <property name="pixmap">
        <pixmap resource="../../data/icons.qrc">:/icons/64x64/spotify.png</pixmap>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>LoginStateWidget</class>
   <extends>QWidget</extends>
   <header>widgets/loginstatewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>enable</tabstop>
  <tabstop>username</tabstop>
  <tabstop>password</tabstop>
  <tabstop>button_login</tabstop>
  <tabstop>quality</tabstop>
  <tabstop>searchdelay</tabstop>
  <tabstop>artistssearchlimit</tabstop>
  <tabstop>albumssearchlimit</tabstop>
  <tabstop>songssearchlimit</tabstop>
  <tabstop>checkbox_download_album_covers</tabstop>
  <tabstop>checkbox_fetchalbums</tabstop>
  <tabstop>coversize</tabstop>
 </tabstops>
 <resources>
  <include location="../../data/data.qrc"/>
  <include location="../../data/icons.qrc"/>
 </resources>
 <connections/>
</ui>
